src/utils.js

```diff
@@ -63,16 +65,86 @@ async function checkForPlaystationDirectRedirect(checkInterval, onSuccess, versi
         onSuccess();
     } else {
         setTimeout(() => {
-            console.log("No stock detected yet. Scanning again...");
-            console.log("Number of tries", numTries);
-            console.log("");
+            console.log("PS Direct: No queue detected yet. Scanning again...");
+            console.log(`Number of tries: ${numTries}\n`);
             numTries++;
 
             checkForPlaystationDirectRedirect(checkInterval, onSuccess, version, browser, numTries);
         }, checkInterval);
     }
 }
 
+/*  Recursively checks walmart mobile API for online and inStore stock status
+    storeId param based on user input -- set to 1 if no user input, will only
+    check online stock
+*/
+
+async function watchWalmart(checkInterval, onWalmartSuccess, version, storeId, numWalmartTries = 1) {
+    let walmartStockStatus = await checkWalmartStock(version, storeId);
+    let walmartRetry;
+
+    switch(walmartStockStatus) {
+        case "walmartConnectionError":
+            console.log("Walmart Connection Error");
+            console.log(`Number of tries: ${numWalmartTries}\n`);
+            walmartRetry = true;
+            break;
+        case "walmartOnlineFound":
+            console.log("Found Walmart Online Inventory!");
+            onWalmartSuccess();
+            break;
+        case "walmartInStoreFound":
+            console.log("Found Walmart Store Pickup Inventory!");
+            onWalmartSuccess();
+            break;
+        case "noPage":
+            console.log("Product page not up yet.");
+            console.log(`Walmart Number of tries: ${numWalmartTries}`);
+            walmartRetry = true;
+            break;
+        default:
+            console.log("Walmart: No inventory detected yet.  Scanning again...");
+            console.log(`Walmart Number of tries: ${numWalmartTries}\n`);
+            walmartRetry = true;
+            break;
+    }
+    if (walmartRetry == true) {
+        numWalmartTries++;
+        setTimeout(() => {
+            watchWalmart(checkInterval, onWalmartSuccess, version, storeId, numWalmartTries);
+        }, checkInterval);
+    }
+}
+
+/*  Promise to return stock status.  */
+function checkWalmartStock(version, storeId) {
+    return new Promise(async (resolve, reject) => {
+        let testUrl = `https://putsreq.com/dp9X62F4g5Fj37MJ6aX0?version=${version}?storeId=${storeId}`;
+        let prodUrl = `https://search.mobile.walmart.com/v1/products-by-code/UPC/${version}?storeId=${storeId}`;
+        superagent.get(prodUrl)
+            .end((err, res) => {
+                if (err) {
+                    if (res.status == 444) {
+                        return resolve("noPage");
```

Comment:
I think this should work without the `return`. You should be able to just use `resolve()` by itself.